web-server/http-service-impl/src/server.rs

```diff
@@ -0,0 +1,34 @@
+use actix::{Actor, Addr, Arbiter};
+use actix_web::{web, App, HttpServer};
+
+use crate::actors::WalletEnclaveActor;
+use crate::resources;
+use crate::traits::WalletEnclave;
+
+#[derive(Clone)]
+pub(crate) struct AppState {
+    pub(crate) wallet_enclave_addr: Addr<WalletEnclaveActor>,
+}
+
+/// Run the server on the given address.
+pub async fn run_server(
+    wallet_enclave: Box<dyn WalletEnclave>,
+    bind_addr: &str,
+) -> std::io::Result<()> {
+    let enclave_arbiter = Arbiter::new();
+    let wallet_enclave_addr = Actor::start_in_arbiter(&enclave_arbiter.handle(), |_ctx| {
+        WalletEnclaveActor { wallet_enclave }
+    });
+
+    let server = HttpServer::new(move || {
```

Comment:
Does `move` change the port if it's taken?

Reply:
Nah, this is just for data ownership: the closure gets saved and used to spawn new handlers, so it has to own the values it refers to (particularly `wallet_enclave_addr`).